docs/design_docs/011-historacle-pricing.md

```diff
@@ -0,0 +1,134 @@
+
+These values are stored in state in order to avoid the `x/leverage` module from having to calculate them while making decisions around allowable positions for users to take.
+
+### Proposed API
+
+The `x/leverage` module will have access to the following `keeper` functions from the `x/oracle` module:
+- `HistoricMedian(denom) (sdk.Dec, error)` returns the median price of an asset in the last `Pruning Period`
+- `WithinHistoricDeviation(denom) (bool, error)` returns whether or not the current price of an asset is within the `Standard Deviation around the Median`.
+
+### Outcomes
+
+There are a few outcomes to consider:
+
+> This will require a chain upgrade & migration of the `x/oracle` module.
+> In order to balance performance and safety, we will have to debate the initial governance parameters.
+
+## Alternative Approaches
+
+### Feeder implementation
+
+This implementation takes the on-chain process of storing historic prices and calculating the median, and takes it off-chain. Validators would then vote on the prices of these assets in a similar fashion to the `x/oracle` voting process.
+
+There are a couple reasons for not doing this:
+
+1. Technical overhead - the x/oracle module would either have to be heavily augmented to deal with this, or we would have to build a new module to deal with the voting process, which would lead to code replication.
+2. Latency - validators take time to come to consensus in this model. Currently prices are 5 blocks behind the current APIs, this would make the median another 5 blocks behind that.
+
+## Consequences
+
+The major negative consequence of this is state bloat. Keeping the prices, medians, and standard deviations of each asset supported by the oracle module leads to a lot of bloat.
+
+The major positive is less technical overhead by utilizing an existing module, and allowing for more safety features for `x/leverage`.
+
+### Backwards Compatibility
+
+This will not introduce a new module, and it is relatively backwards compatible. We should allow for governance to disable this feature by voting to set `Stamp Period`, `Prune Period`, and `Median Period` to 0.
+
+### Positive
+
+- Efficient API for the `x/leverage` module to use for safety.
+- We can continue listing low-volume assets for collateral.
+- Use of an existing module rather than creating a new one.
+- Protects against "borrowing + price dump" attacks which are not prevented by disabling low-volume asset collateral
+
+### Negative
+
+- State bloat.
+- Additional processing to do during each epoch.
+
+## Further Discussions
+
+1. Can we lessen state bloat without impacting the `x/leverage` module's efficiency?
+
+## Comments
+
+Originally a 24-hour median was thought to be effective enough to defend against this type of attack, but our modeling shows that we would need 30 days for adequate protection.
```

Comment:
where is this model? It sounds that 30 days is really a long. The price may not reflect the reality. Reality is _now_ modulo possible attack.